src/Dfc.CourseDirectory.WebV2/Features/ProviderSearch/ProviderSearch.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.Core.Models;
+using Dfc.CourseDirectory.Core.Search;
+using Dfc.CourseDirectory.WebV2.Security;
+using MediatR;
+using OneOf;
+using OneOf.Types;
+using SearchModels = Dfc.CourseDirectory.Core.Search.Models;
+
+namespace Dfc.CourseDirectory.WebV2.Features.ProviderSearch
+{
+    public class Query : IRequest<ViewModel>
+    {
+        public string SearchQuery { get; set; }
+    }
+
+    public class ViewModel : Query
+    {
+        public IReadOnlyCollection<ProviderSearchResultViewModel> Providers { get; set; }
+    }
+
+    public class ProviderSearchResultViewModel
+    {
+        public Guid ProviderId { get; set; }
+        public string Name { get; set; }
+        public string Postcode { get; set; }
+        public string Town { get; set; }
+        public string UKPRN { get; set; }
+        public ProviderStatus? Status { get; set; }
+        public string ProviderStatus { get; set; }
+    }
+
+    public class OnboardProviderCommand : IRequest<OneOf<NotFound, Success>>
+    {
+        public Guid ProviderId { get; set; }
+    }
+
+    public class Handler :
+        IRequestHandler<Query, ViewModel>,
+        IRequestHandler<OnboardProviderCommand, OneOf<NotFound, Success>>
+    {
+        private readonly ISearchClient<SearchModels.Provider> _providerSearchClient;
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ICurrentUserProvider _currentUserProvider;
+        private readonly IClock _clock;
+
+        public Handler(
+            ISearchClient<SearchModels.Provider> providerSearchClient,
+            ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
+            ICurrentUserProvider currentUserProvider,
+            IClock clock)
+        {
+            _providerSearchClient = providerSearchClient;
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _currentUserProvider = currentUserProvider;
+            _clock = clock;
+        }
+
+        public async Task<ViewModel> Handle(Query request, CancellationToken cancellationToken)
+        {
+            if (request.SearchQuery == null)
```

Comment:
Is it worth making the no search/no results/some results variants of the view model more explicit? It's not super obvious to me looking at the `ViewModel` signature that `null` means 'no search was done' vs an empty collection meaning 'there are no results'.

Reply:
I did ponder this.... best I could come up with was a boolean `PerformSearch` (or similar) but I think that was less preferable to the existing use of nulls. Any suggestions?

Reply:
The `FindStandard` search sets a `SearchWasDone` flag on the VM but I'm not crazy about that naming. (`SearchPerformed` maybe?) Alternatively something like `OneOf<Nothing, QueryResponse>`, but you'd have to create the `Nothing` type.

Reply:
As discussed, couldn't find a nicer way to make this work in code so have added some comments around behaviours.